service/controller/v21/resource/cloudformation/current.go

```diff
@@ -217,22 +216,53 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 			return StackState{}, microerror.Mask(err)
 		}
 
-		workerCount, err := ctlCtx.CloudFormation.GetOutputValue(stackOutputs, key.WorkerCountKey)
+		workerASGName, err := ctlCtx.CloudFormation.GetOutputValue(stackOutputs, key.WorkerASGKey)
 		if err != nil {
 			return StackState{}, microerror.Mask(err)
 		}
-		workerImageID, err := ctlCtx.CloudFormation.GetOutputValue(stackOutputs, key.WorkerImageIDKey)
+		workerCloudConfigVersion, err := ctlCtx.CloudFormation.GetOutputValue(stackOutputs, key.WorkerCloudConfigVersionKey)
 		if err != nil {
 			return StackState{}, microerror.Mask(err)
 		}
-		workerInstanceType, err := ctlCtx.CloudFormation.GetOutputValue(stackOutputs, key.WorkerInstanceTypeKey)
+		workerImageID, err := ctlCtx.CloudFormation.GetOutputValue(stackOutputs, key.WorkerImageIDKey)
 		if err != nil {
 			return StackState{}, microerror.Mask(err)
 		}
-		workerCloudConfigVersion, err := ctlCtx.CloudFormation.GetOutputValue(stackOutputs, key.WorkerCloudConfigVersionKey)
+		workerInstanceType, err := ctlCtx.CloudFormation.GetOutputValue(stackOutputs, key.WorkerInstanceTypeKey)
 		if err != nil {
 			return StackState{}, microerror.Mask(err)
 		}
+		var workerDesired, workerMin, workerMax int
+		{
+			r.logger.LogCtx(ctx, "level", "debug", "message", "finding out worker asg properties")
+
+			i := &autoscaling.DescribeAutoScalingGroupsInput{
+				AutoScalingGroupNames: []*string{
+					&workerASGName,
+				},
+			}
+			o, err := ctlCtx.AWSClient.AutoScaling.DescribeAutoScalingGroups(i)
+			if err != nil {
+				return StackState{}, microerror.Mask(err)
+			}
+			if len(o.AutoScalingGroups) == 0 {
+				return StackState{}, microerror.Maskf(notExistsError, "asg for name %s", workerASGName)
+			}
+			asg := o.AutoScalingGroups[0]
```

Comment:
Should we handle case when `len(o.AutoSaclingGroups) > 1` or `!= 1` in general?

Reply:
Not yet. Time for that will come when we start working on node pools, but that will change many places so radically any way that I don't consider changes related to that right now. There's only one worker ASG and it's selected by ARN so length of results is either 0 or 1.